packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -1,15 +1,80 @@
-import * as React from "react";
-import { CustomCell, CustomRenderer, drawTextCell, GridCellKind } from "@glideapps/glide-data-grid";
+import React from "react";
+import { assertNever } from "../../../core/src/common/support";
+import {
+    CustomCell,
+    CustomRenderer,
+    drawTextCell,
+    GridCellKind,
+    ProvideEditorCallback,
+} from "@glideapps/glide-data-grid";
 
 interface DatePickerCellProps {
     readonly kind: "date-picker-cell";
     readonly date: Date | undefined;
     readonly displayDate: string;
-    readonly format: "date" | "datetime-local";
+    readonly format: DateType;
 }
 
+export type DateType = "date" | "time" | "datetime-local";
+
+export const formatValueForHTMLInput = (format: DateType, date: Date): string => {
+    switch (format) {
+        case "date":
+            return date.toISOString().split("T")[0];
+        case "datetime-local":
+            return date.toISOString().replace("Z", "");
+        case "time":
+            return date.toISOString().split("T")[1].replace("Z", "");
+        default:
+            assertNever(format);
+    }
+};
+
 export type DatePickerCell = CustomCell<DatePickerCellProps>;
 
+const Editor: ReturnType<ProvideEditorCallback<DatePickerCell>> = cell => {
+    const cellData = cell.value.data;
+    const { date, displayDate, format } = cellData;
+    const cellDataDate =
+        cellData.date ??
+        // now as date with adjusted timezone to allow for displayDate to be correct
+        new Date(new Date().getTime() - new Date().getTimezoneOffset() * 60000);
```

Comment:
I think there are also two options here in case cell value is `undefined` and user opens the editor:
1) Initializing the value of the input with the user's current date/time (this implementation)
2) Setting the value of the input to empty/undefined -> this would show the empty template of the input, e.g.:
<img width="187" alt="image" src="https://user-images.githubusercontent.com/2852129/215956182-61a1f615-c093-49e4-a7f8-74aa80e154ac.png">

My preference would be option 2 here since it is a bit more consistent with other existing cell editors showing the actual current value of the cell.

Reply:
That's a good point. I'll change it to undefined.

Reply:
Thanks Lukas :)